charmcraft/snap.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Logic for snap configuration."""
+
+import logging
+from typing import Optional
+
+import attr
+import snaphelpers
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
```

Comment:
why aren't you using dataclasses now?

Reply:
In this case, either works just fine. I had just grepped the codebase and found only one instance of attr usage, but no dataclasses.  Just for consistency I stuck with attr.

If we want to standardize on dataclass here, I'll be happy to change this (and the other instance).

Reply:
attrs was used at the time only because of the runtime dependency. I would change here, the other one in a follow up

Reply:
Yes, we used `attrs` because of dataclasses were not supported in 3.5. Let's go full dataclasses!

Reply:
:+1:

Reply:
other change done @ https://github.com/canonical/charmcraft/pull/483